Remove /auth from URL path

thalman opened the pull request:
New keycloak (that uses quarkus) runs on web root.

 This PR removes /auth from the path. The keycloak_httpd_client
 can be still used with the old version, just the --server parameter
 must be complete URL (https://keycloak.example.com/auth)

thalman committed:
Remove /auth from URL path

New keycloak (that uses quarkus) runs on web root.

This commit removes /auth from the path. The keycloak_httpd_client
can be still used with the old version, just the --server parameter
must be complete URL (https://keycloak.example.com/auth)

thalman committed:
Update --server parameter help

The parameter is in fact URL, not server name or IP address.

thalman commented:
This PR address the same issue as #7, it just uses different approach.

thalman commented:
Hi @spoore1,

It is possible to write some auto detection for fallback to `/auth` URI for old keycloak. I'm just not sure if this is worth the effort. WDYT?

Tom

spoore1 commented:
Looking at it, it seems pretty straightforward and simple.   When I run some tests with /auth added, everything works as expected.   When I run it using the default without /auth, the oauth test fails.  I'm looking into this.

thalman committed:
Strip trailing / from server URL

It is convinient for user to copy URL and it may contain
trailing slash. With this change trailing slash is removed
so we do not produce URLs with double-slash when formating.

thalman committed:
Auto-detect /auth prefix in URL

Old keycloak (16) used /auth path in the url. With this update
we try to add /auth to the url and we test if it works.

thalman commented:
> Looking at it, it seems pretty straightforward and simple. When I run some tests with /auth added, everything works as expected. When I run it using the default without /auth, the oauth test fails. I'm looking into this.

I fixed this, tested in my environment.

thalman commented:
I added detection if the old prefix "/auth" is present.

spoore1 commented:
I still see the oauth test failing with the new change.

This is what I'm trying:

- build rpm from this branch
- install rpm
- git clone https://github.com/latchset/federation_testing
- cd federation_testing
- vim setup.sh
    - Remove all keycloak server url references with /auth
- vim test_oidc.sh
    - Remove all keycloak server url references with /auth
- sh -x setup.sh
- sh -x test_oidc.sh

When I run these without the change to remove /auth references, everything passes as expected.  It's only when I remove the /auth references that the test_oauth test case fails.   I'm not quite sure yet why.

spoore1 commented:
@thalman the suggested change to oidctest.py that I missed in the upstream test did fix the issue I had with the oauth test:

```
test_oidc.py::test_oauth PASSED                                                                 [ 75%]
```

spoore1 reviewed (approved):
With test issue resolved, this LGTM.
